sklearn/metrics/classification.py

```diff
@@ -179,6 +179,80 @@ def accuracy_score(y_true, y_pred, normalize=True, sample_weight=None):
     return _weighted_sum(score, sample_weight, normalize)
 
 
+def top_n_accuracy_score(y_true, y_pred, n=5, normalize=True):
+    """top N Accuracy classification score.
+    For multiclass classification tasks, this metric returns the
+    number of times that the correct class was among the top N classes
+    predicted.
+
+    Parameters
+    ----------
+    y_true : 1d array-like, or label indicator array / sparse matrix
+        Ground truth (correct) labels.
+
+    y_pred : array-like, where for each sample, each row represents the
+        likelihood of each possible label.
+        The number of columns must be at least as large as the set of possible
+        label values.
+
+    normalize : bool, optional (default=True)
+        If ``False``, return the number of top N correctly classified samples.
+        Otherwise, return the fraction of top N correctly classified samples.
+
+    Returns
+    -------
+    score : float
+        If ``normalize == True``, return the proportion of top N correctly
+        classified samples, (float), else it returns the number of top N
+        correctly classified samples (int.)
+
+        The best performance is 1 with ``normalize == True`` and the number
+        of samples with ``normalize == False``.
+
+    See also
+    --------
+    accuracy_score
+
+    Notes
+    -----
+    If n = 1, the result will be the same as the accuracy_score. If n is the
+    same as the number of classes, this score will be perfect and meaningless.
+
+    In cases where two or more classes are assigned equal likelihood, the
+    result may be incorrect if one of those classes falls at the threshold, as
+    one class must be chosen to be the nth class and the class chosen may not
+    be the correct one.
```

Comment:
Incorrect - as in slightly inaccurate or completely wrong? Is there a way around it?

Reply:
Incorrect. Consider a single observation whose target is class 1 (out of 5). Suppose the predicted probabilities are y_pred = [0.4, 0.2, 0.1, 0.2, 0.1]. The top 3 predictions are 0, 1, and 3, so the top-3 accuracy is 1.0 on this observation. The top-2 predictions are 0 and 3 (np.argsort returns indices containing equal values in ascending order, and this metric takes the tail of the argsort list), so the top-2 accuracy is 0 on this observation: even though classes 1 and 3 are assigned the same probabilities, class 1 doesn't make the cut. I could replace the disclaimer with logic to address this - in, tensorflow, for instance, any class assigned probability equal to a probability that made the top-n list to be in the top n. With that change the top-2 prediction in the example described would be the same as the top-3 accuracy. Let me know what you think.